pkg/purge/purge.go

```diff
@@ -0,0 +1,221 @@
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+type PurgeClient struct {
+	clientConfig *rest.Config
+	mapper       meta.RESTMapper
+	client       dynamic.Interface
+}
+
+type PurgeOptions struct {
+	Type                    bootstrap.BootstrapType
+	Url                     string
+	Latest                  bool
+	CreateDefaultRepository bool
+	DisableTelemetry        bool
+	Force                   bool
+}
+
+func DefaultPurgeOptions() PurgeOptions {
+	return PurgeOptions{Type: bootstrap.BootstrapTypeAio, Latest: config.IsDevBuild(), CreateDefaultRepository: true}
+}
+
+func NewPurgeClient(config *rest.Config) *PurgeClient {
+	return &PurgeClient{clientConfig: config}
+}
+
+func (c *PurgeClient) initRestMapper() error {
+	if discoveryClient, err := discovery.NewDiscoveryClientForConfig(c.clientConfig); err != nil {
+		return err
+	} else if groupResources, err := restmapper.GetAPIGroupResources(discoveryClient); err != nil {
+		return err
+	} else {
+		c.mapper = restmapper.NewDiscoveryRESTMapper(groupResources)
+		return nil
+	}
+}
+
+func (c *PurgeClient) Purge(ctx context.Context, options PurgeOptions) error {
+	if err := c.initRestMapper(); err != nil {
+		return err
+	}
+
+	if client, err := dynamic.NewForConfig(c.clientConfig); err != nil {
+		return err
+	} else {
+		c.client = client
+	}
+
+	fmt.Println("Starting purge process")
+	start := time.Now()
+
+	if options.Url == "" {
+		version := config.Version
+		if options.Latest {
+			if releaseInfo, err := releaseinfo.FetchLatestRelease(); err != nil {
+				if httperror.Is(err, http.StatusServiceUnavailable) || httperror.IsTimeoutError(err) {
+					telemetry.BootstrapFailure(time.Since(start))
+					return fmt.Errorf("network connectivity error, check your network: %w", err)
+				}
+				telemetry.BootstrapFailure(time.Since(start))
+				return fmt.Errorf("could not determine latest version: %w", err)
+			} else {
+				version = releaseInfo.Version
+			}
+		}
+		options.Url = fmt.Sprintf("https://github.com/glasskube/glasskube/releases/download/v%v/manifest-%v.yaml",
+			version, options.Type)
+	}
+
+	statusMessage("Fetching Glasskube manifest from "+options.Url, true)
+	manifests, err := clientutils.FetchResources(options.Url)
+	if err != nil {
+		statusMessage("Couldn't fetch Glasskube manifests", false)
+		return fmt.Errorf("couldn't fetch Glasskube manifests: %w", err)
+	}
+
+	statusMessage("Validating existing installation", true)
+
+	if err = c.preprocessManifests(ctx, manifests, &options); err != nil {
+		telemetry.BootstrapFailure(time.Since(start))
+		statusMessage(fmt.Sprintf("Couldn't prepare manifests: %v", err), false)
+		if !options.Force {
+			return err
+		} else {
+			statusMessage("Attempting to force bootstrap anyways (Force option is enabled)", true)
+		}
+	}
+
+	if options.CreateDefaultRepository {
+		manifests = append(manifests, bootstrap.DefaultRepository())
+	}
```

Comment:
Purge deletes the `PackageRepository` CRD anyways, so deleting the default repository should not be necessary.